pkgs/tools/virtualization/guestfs-tools/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchurl
+, bison
+, cdrkit
+, cpio
+, flex
+, getopt
+, hivex
+, jansson
+, libguestfs
+, libvirt
+, libxml2
+, makeWrapper
+, ncurses
+, ocamlPackages
+, pcre2
+, perlPackages
+, pkg-config
+, qemu
+, xz
+}:
+
+stdenv.mkDerivation rec {
+  pname = "guestfs-tools";
+  version = "1.48.2";
+
+  nativeBuildInputs = [
+    bison
+    cdrkit
+    cpio
+    flex
+    getopt
+    makeWrapper
+    pkg-config
+    qemu
+  ] ++
+  (with perlPackages; [
+    GetoptLong
+    libintl-perl
+    ModuleBuild
+    perl
+    Po4a
+  ]) ++
+  (with ocamlPackages; [
+    findlib
+    gettext-stub
+    ocaml
+    ocaml_gettext
+    ounit2
+  ]);
+
+  buildInputs = [
+    hivex
+    jansson
+    libguestfs
+    libvirt
+    libxml2
+    ncurses.dev
+    pcre2.dev
+    xz.dev
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/virt-win-reg \
+      --prefix PERL5LIB : ${with perlPackages; makeFullPerlPath [ hivex libintl-perl libguestfs ]}
+  '';
+
+  enableParallelBuilding = true;
+
+  preBuild = ''
+    patchShebangs .
+  '';
+
+  src = fetchurl {
+    url = "https://download.libguestfs.org/${pname}/${lib.versions.majorMinor version}-stable/${pname}-${version}.tar.gz";
+    sha256 = "sha256-G9l5sG5g5kMlSXzg0GX8+Et7M9/k2dRLMBgsMI4MaxA=";
+  };
```

Comment:
I think this section should be at the top to follow Nixpkgs conventions.

Reply:
I don't think we have any convention about formatting derivations (at least I never saw one, if you can link one to me I can fix this).

Reply:
Well, now that I am looking again the order of this derivation is weird. Let me fix this.

Reply:
Done.

Reply:
Sorry, I wasn't specific on this. I just wanted to say that pretty much all of other packages in Nixpkgs have the `src` block right after the `version`. I think it would be better to follow this so that peoples' expectations are met regarding the location of that block. Well, at least I look for `src` there.